### PR TITLE
[#2362] improvement: enable `UseCorrectAssertInTests` error-prone check

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

Enable `UseCorrectAssertInTests` error-prone check to avoid using `assert` in tests.

### Why are the changes needed?

Error prone has `UseCorrectAssertInTests` bug pattern that Java assert is used in testing code. For testing purposes, prefer using Truth-based assertions in [UseCorrectAssertInTests](https://errorprone.info/bugpattern/UseCorrectAssertInTests). Therefore enable `UseCorrectAssertInTests` check to avoid using `assert` in tests.

Fix: #2362

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

`./gradlew build`

### CODE REVIEW

**Comment by SteNicholas** [comment]
cc @yuqi1129, @jerryshao.

**Comment by jerryshao** [comment]
@coolderli please also help to review.

File: api/src/main/java/com/datastrato/gravitino/rel/TableChange.java [hunk 1]
```diff
@@ -581,9 +583,7 @@ final class After implements ColumnPosition {
     private final String column;
 
     private After(String column) {
-      if (null == column) {
-        throw new IllegalArgumentException("column can not be null.");
-      }
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(column), "Column cannot be null");
```
Comment (jerqi) [inline api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:586]: Is this an unrelated change about `UseCorrectAssertInTests `?

Comment (qqqttt123) [inline api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:586]: Is this an unrelated change about `UseCorrectAssertInTests`?

File: catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/utils/DataSourceUtils.java [hunk 2]
```diff
@@ -65,12 +66,10 @@ private static Properties getProperties(JdbcConfig jdbcConfig) {
   public static void closeDataSource(DataSource dataSource) {
     if (null != dataSource) {
       try {
-        if (dataSource instanceof BasicDataSource) {
-          ((BasicDataSource) dataSource).close();
-        } else {
-          throw new UnsupportedOperationException(
-              "close operation can only be called in BasicDataSource.");
-        }
+        Preconditions.checkArgument(
```
Comment (qqqttt123) [inline catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/utils/DataSourceUtils.java:69]: ditto.

File: api/src/main/java/com/datastrato/gravitino/rel/TableChange.java (hunk 1 above)
Comment (SteNicholas) [reply api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:586]: @qqqttt123, #2362 mentions this change as `We'd better using Guava's Preconditions, rather than if..else..., which is not elegant.`. Meanwhile, I have mentioned this change in description.

Comment (qqqttt123) [reply api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:586]: I prefer not including this change in this pr.

Comment (yuqi1129) [reply api/src/main/java/com/datastrato/gravitino/rel/TableChange.java:586]: I'm okay with it.

**Comment by yuqi1129** [comment]
@SteNicholas 
Could you help to check all possible `if/else` code blocks that check arguments and optimize them all?

**Comment by SteNicholas** [comment]
@yuqi1129, I will push another pull request for this improvement of if/else code blocks. This pull request solves the `UseCorrectAssertInTests`.

**Review by qqqttt123 — APPROVED** [review]
LGTM. thanks @SteNicholas for contribution. thanks @yuqi1129 for review.

**Comment by jerryshao** [comment]
LGTM, thanks @SteNicholas for your code.
